includes/class-indieauth-authenticate.php

```diff
@@ -92,29 +96,33 @@ public static function authorization_redirect( $me, $redirect_uri ) {
 		$query = build_query(
 			array(
 				'me'           => rawurlencode( $me ),
-				'redirect_uri' => wp_login_url( $redirect_uri ),
-				'client_id'    => home_url(),
+				'redirect_uri' => rawurlencode( $redirect_uri ),
+				'client_id'    => rawurlencode( home_url() ),
 				'state'        => wp_create_nonce( 'indieauth-' . home_url() ),
```

Comment:
This needs to store the generated nonce here so that it can be compared later when redirected back. Usually this is stored in the server session, not sure how Wordpress wants you to do it.